core/src/main/scala/caliban/parsing/VariablesUpdater.scala

```diff
@@ -16,38 +21,75 @@ object VariablesUpdater {
   ): GraphQLRequest = {
     val variableDefinitions = doc.operationDefinitions.flatMap(_.variableDefinitions)
     val updated             = req.variables.getOrElse(Map.empty).map { case (key, value) =>
-      val v = variableDefinitions.find(_.name == key).map(resolveEnumValues(value, _, rootType)).getOrElse(value)
+      val v =
+        variableDefinitions
+          .find(_.name == key)
+          .map { (definition: VariableDefinition) =>
+            rewriteValues(value, definition.variableType, rootType)
+          }
+          .getOrElse(value)
+
       key -> v
     }
 
     req.copy(variables = Some(updated))
   }
 
+  def rewriteValues(value: InputValue, `type`: Type, rootType: RootType): InputValue =
+    `type` match {
+      case ListType(ofType, nonNull) =>
```

Comment:
You can use wildcards for unused values like `nonNull`

Reply:
Hm, VSCode usually warns me about that, but maybe that's only if `-Wunused:params` is enabled.